replica/backup.go

```diff
@@ -175,24 +180,92 @@ func (rb *Backup) findIndex(id string) int {
 	return -1
 }
 
+type Hole struct {
+	f      types.DiffDisk
+	offset int64
+	len    int64
+}
+
+var HoleCreatorChan = make(chan Hole, 1000000)
+
+//CreateHoles removes the offsets from corresponding sparse files
+func CreateHoles() error {
+	retryCount := 0
+	for {
+		hole := <-HoleCreatorChan
+	retry:
+		if err := syscall.Fallocate(int(hole.f.Fd()),
+			sparse.FALLOC_FL_KEEP_SIZE|sparse.FALLOC_FL_PUNCH_HOLE,
+			hole.offset*defaultSectorSize, hole.len*4096); err != nil {
```

Comment:
here, offset and len should be multiplied by d.sectorsize.. let us pass sectorsize also in the Hole data structure or multiply it and add it to the channel